feat: add liens capitalistiques

rmonnier9 opened the pull request:
- Fonctionnalité
- Détails :
  - Ajout des liens capitalistiques de la DGFIP

Closes #1641 

<img width="931" alt="image" src="https://github.com/user-attachments/assets/13b4deed-ab1d-4f2b-b52f-8928ffaf8590" />

<img width="1280" alt="image" src="https://github.com/user-attachments/assets/b3af2fe6-5650-457f-b321-4be6dafcbeda" />

<img width="1257" alt="image" src="https://github.com/user-attachments/assets/4d3191b3-e764-4cf0-9396-de91cbd96387" />

<img width="1245" alt="image" src="https://github.com/user-attachments/assets/63ae7bfd-68d0-4f88-abd2-4f21d3c377a0" />

XavierJp commented:
I would say this lacks explanations

- You could be more explicit about the origin of the data as it explain why it is accessed by year and why last year lacks data
- You could be more explicit about actionnaires end filiales. Without clear definition, we dont immediatyl understand who owns who
- I think reusing the cerfa title is smart as people might use the terms 
<img width="592" alt="image" src="https://github.com/user-attachments/assets/3337efb5-c229-40c6-a768-d3ce140b55be" />
- we could add more variables for instance, country would be a nice addition when not FR

https://entreprise.api.gouv.fr/catalogue/dgfip/liens_capitalistiques

XavierJp commented on app/(header-default)/dirigeants/[slug]/_component/liens-capitalistiques/protected-liens-capitalistiques-section.tsx:
```diff
@@ -0,0 +1,184 @@
+    params
+  );
+
+  if (isDataLoading(liensCapitalistiquesProtected)) {
+    return <Loader />;
+  }
+
```
Same issue as with `<LiassesFiscales />` once a year has been selected, `<Loader />` is not displayed anymore when we change year

rmonnier9 replied:
> Same issue as with `<LiassesFiscales />` once a year has been selected, `<Loader />` is not displayed anymore when we change year

I've fixed it for both :)

XavierJp commented on cypress/fixtures/dgfip-liens-capitalistiques.json:
```diff
@@ -0,0 +1,70 @@
+  "data": {
+    "capital": {
+      "actionnaires": [
+        {
+          "type": "personne_morale",
+          "pourcentage": 51,
+          "nombre_parts": 510,
+          "personne_physique_attributes": {
+            "civilite": "M",
+            "nom_patronymique_et_prenoms": "CHIRAC Jacques",
+            "nom_marital": "NS",
+            "date_naissance": {
+              "annee": "1932",
+              "mois": "11"
+            }
+          },
+          "personne_morale_attributes": {
+            "siren": "110001013",
+            "denomination": "GOUVERNEMENT PREMIER MINISTRE",
```
I love it

XavierJp reviewed (changes requested):
Clean and simple !

rmonnier9 commented:
> I would say this lacks explanations
> 
> * You could be more explicit about the origin of the data as it explain why it is accessed by year and why last year lacks data

I've added more informations while trying to be concise.

> * You could be more explicit about actionnaires end filiales. Without clear definition, we dont immediatyl understand who owns who

I've added a title before each table.

> * you can use several column, no need to merge everything. Especially as siren and form juridique have constant length. Plus a column label add a contextual explanation when SA or NS is not self explanatory

I've done it for "Liste des participations" but for "Composition du capital" it can also be "Personne physique" so I've reused the same display as we were doing for "Dirigeants" section.

> * country would be a nice addition when not FR

Indeed, good idea.

> * I think reusing the cerfa title is smart as people might use the terms

RIght !

rmonnier9 commented:
The select now takes into account the company creation date in order to only display years where there is a "liasse fiscale".

A company created in 2022 :
<img width="1319" alt="image" src="https://github.com/user-attachments/assets/d298c8cc-46cc-4341-a5f2-999f29861f1b" />

rmonnier9 commented:
I've added the country for each time, maybe it could only be displayed when the "personne morale/physique" is not french

XavierJp commented:
Very nice changes, I love it ! I refactored just a bit the text organisation and added placeholder when no actionnaires or no filiales

XavierJp commented on app/(header-default)/dirigeants/[slug]/_component/sections/entreprise/summary.tsx:
```diff
@@ -26,6 +26,9 @@ const DirigeantSummary: React.FC<IDirigeantSummaryProps> = ({
             (RNE)
           </a>
         </li>
+        <li>
+          <a href="#liens-capitalistiques">Liste des liens capitalistiques</a>
+        </li>
```
You might need a condition on this. Not sure it should display for everyone

XavierJp commented on app/(header-default)/dirigeants/[slug]/_component/liens-capitalistiques/index.tsx:
```diff
@@ -0,0 +1,29 @@
+      title="Liens capitalistiques"
+      id="liens-capitalistiques"
+      sources={[EAdministration.DGFIP]}
+      allowedUseCases={[UseCase.fraude]}
```
I think it is allowed for more useCases

XavierJp replied:
https://entreprise.api.gouv.fr/catalogue/dgfip/liens_capitalistiques

XavierJp reviewed (approved):
A few minor fixes and ready to ship !
